docroot/profiles/custom/cgov_site/modules/custom/cgov_core/src/Plugin/Block/PageOptions.php

```diff
@@ -60,8 +61,23 @@ protected function blockAccess(AccountInterface $account) {
   public function build() {
     return [
       '#type' => 'block',
-      'options' => $this->pageOptionsConfigs,
+      'options' => $this->pageOptionsConfigs
     ];
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function getCacheMaxAge() {
+    return 0;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getCacheContexts() {
+    return Cache::mergeContexts(parent::getCacheContexts(), ['url.path']);
```

Comment:
Yes , we can remove the cache context since we are setting MaxAge as 0. We set this to 0 because each page the content is dynamic. But QA raised defect while doing regression. We analysed and found that block content is showing wrong for each page because of caching.

Reply:
This is done